test/unit/public-sdk-apis/setExternalUserIdWithSecondaryChannel.ts

```diff
@@ -87,3 +87,22 @@ test("setExternalUserId before email, includes external_user_id in POST create c
     TEST_EXTERNAL_USER_ID
   );
 });
+
+test("setExternalUserId after email, w/o push player, makes PUT call to update Email record", async t => {
+  // 1. Nock out email create
+  const emailPostNock = NockOneSignalHelper.nockPlayerPost();
+  await OneSignal.setEmail(TEST_EMAIL_ADDRESS);
+  const emailPlayerId = (await emailPostNock.result).response.body.id;
+
+  // 2. Call OneSignal.setExternalUserId and ensure email is updated
+  const externalUserIdUpdateOnEmail = NockOneSignalHelper.nockPlayerPut(emailPlayerId);
+  await OneSignal.setExternalUserId(TEST_EXTERNAL_USER_ID);
+
+  t.deepEqual(
```

Comment:
add a comment noting what is different in the body (why is this failing prior to the change?)

Reply:
👍 I'll add a comment

Reply:
Done